Patch by Robert Scheck <robert@fedoraproject.org> for prosody >= 0.10 which removes all
options that OpenSSL 0.9.8e (as shipped by Red Hat Enterprise Linux 5 and derivates such
as CentOS) does not support: SSL_OP_NO_TICKET (added by upstream with OpenSSL 0.9.8f) and
the support for secp384r1 (added by Red Hat for RHEL 6.5 with openssl-1.0.1e-15). LuaSec
0.5 does not offer any option to detect which OpenSSL features are (not) available, thus
Prosody upstream tries to determine them by the LuaSec version - risky, because EPEL 5 is
shipping LuaSec 0.4.1 since ever, but still had no support for the "no_ticket" option.

--- prosody-0.10/core/certmanager.lua			2015-11-04 20:58:26.000000000 +0100
+++ prosody-0.10/core/certmanager.lua.rhel5		2015-11-04 21:02:05.000000000 +0100
@@ -39,7 +39,7 @@
 local luasec_has = {
 	-- TODO If LuaSec ever starts exposing these things itself, use that instead
 	cipher_server_preference = luasec_version >= 2;
-	no_ticket = luasec_version >= 4;
+	no_ticket = nil;
 	no_compression = luasec_version >= 5;
 	single_dh_use = luasec_version >= 2;
 	single_ecdh_use = luasec_version >= 2;
@@ -65,7 +65,6 @@
 		single_ecdh_use = luasec_has.single_ecdh_use;
 	};
 	verifyext = { "lsec_continue", "lsec_ignore_purpose" };
-	curve = "secp384r1";
 	ciphers = "HIGH+kEDH:HIGH+kEECDH:HIGH:!PSK:!SRP:!3DES:!aNULL";
 }
 local path_options = { -- These we pass through resolve_path()
@@ -97,6 +96,7 @@
 	if mode == "server" then
 		if not user_ssl_config.key then return nil, "No key present in SSL/TLS configuration for "..host; end
 		if not user_ssl_config.certificate then return nil, "No certificate present in SSL/TLS configuration for "..host; end
+		if user_ssl_config.options then user_ssl_config.options.no_ticket = nil; end
 	end
 
 	for option in pairs(path_options) do
